File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/renderer/dto/PhpClassInterface.java
package com.github.tobi812.sprykerplugin.models.renderer.dto;

import java.util.List;

public interface PhpClassInterface extends PhpFileInterface {

    Boolean isAbstract();

    void setAbstract();

    PhpClassInterface getParentClass();

    void setParentClass(PhpClassInterface parentClass);

    void setParentClass(PhpClassInterface parentClass, String aliasPrefix);

    List<UseBlockItem> getUseBlockItems();

    void addUseBlockItem(UseBlockItem useBlockItem);

    List<DocBlockItem> getDocBlockItems();

    void addDocBlockItem(DocBlockItem docBlockItem);

    String getParentAlias();


}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/renderer/dto/PhpClassInterface.kt
package com.github.tobi812.sprykerplugin.models.renderer.dto

interface PhpClassInterface : PhpFileInterface {
    val isAbstract: Boolean
    fun setAbstract(): PhpClassInterface

    var parentClass: PhpClassInterface?
    var parentAlias: String?
    fun setParentClass(parentClass: PhpClassInterface): PhpClassInterface
    fun setParentClass(parentClass: PhpClassInterface, aliasPrefix: String): PhpClassInterface

    val useBlockItems: List<UseBlockItem>
    fun addUseBlockItem(useBlockItem: UseBlockItem): PhpClassInterface

    val docBlockItems: List<DocBlockItem>
    fun addDocBlockItem(docBlockItem: DocBlockItem): PhpClassInterface
}